gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/PlaceHolderDiskRegion.java

```diff
@@ -197,4 +207,30 @@ public void initializeStats(long numEntriesInVM, long numOverflowOnDisk,
       this.numOverflowOnDisk.set(numOverflowOnDisk);
       this.numOverflowBytesOnDisk.set(numOverflowBytesOnDisk);
     }
-  }
\ No newline at end of file
+
+    protected long calculateEntryOverhead(RegionEntry entry) {
+      if (entryOverHead == -1L && callback.isSnappyStore()) {
```

Comment:
Do we expect this part of the code to be called more than once? I think this will be by one main thread trying to initialize the PHDR. So sync shouldn't be required.